cmd/notify-mailer/main.go

```diff
@@ -253,7 +295,9 @@ func (m *mailer) resolveAddresses(ctx context.Context) (addressToRecipientMap, e
 				m.log.Errf("Unparsable address %q, skipping ID (%d)", address, recipient.id)
 				continue
 			}
-			result[parsed.Address] = append(result[parsed.Address], recipient)
+			if _, ok := m.readEmailsMap[parsed.Address]; !ok {
+				result[parsed.Address] = append(result[parsed.Address], recipient)
+			}
```

Comment:
Hmm. So while the end result here is what we want -- only updating the map for addresses that we haven't already processed -- this stanza is too late to save us from actually doing extra work. We're still doing all of the database lookups for every recipient, we're just dropping the result on the floor if it turns out we've looked this one up before.

I think this needs to happen earlier. When we read the emails map from disk, we should _also_ keep track of every regID seen in that map. And then we can add something like
```go
_, present := m.cachedRecipients[recipient.id]
if present {
  continue
}
```
between lines 286 and 287 of this function, to actually save ourselves work.